sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -244,17 +249,35 @@ public Aggregation toDruidAggregation(
     final AggregatorFactory theAggFactory;
     switch (args.size()) {
       case 1:
-        theAggFactory = aggregatorType.createAggregatorFactory(aggregatorName, fieldName, null, outputType, null);
+        theAggFactory = aggregatorType.createAggregatorFactory(aggregatorName, fieldName, null, outputType, null, true);
         break;
       case 2:
-        int maxStringBytes;
+        Integer maxStringBytes = getMaxStringBytes(rexNodes.get(1), aggregateCall, plannerContext);
+        if (maxStringBytes == null) {
+          return null;
+        }
+        theAggFactory = aggregatorType.createAggregatorFactory(
+            aggregatorName,
+            fieldName,
+            null,
+            outputType,
+            maxStringBytes.intValue(),
+            true
+        );
+        break;
+      case 3:
+        maxStringBytes = getMaxStringBytes(rexNodes.get(1), aggregateCall, plannerContext);
+        if (maxStringBytes == null) {
+          return null;
+        }
+        boolean aggregateMultipleValues;
         try {
-          maxStringBytes = RexLiteral.intValue(rexNodes.get(1));
+          aggregateMultipleValues = RexLiteral.booleanValue(Preconditions.checkNotNull(rexNodes.get(2)));
         }
-        catch (AssertionError ae) {
+        catch (Exception ae) {
           plannerContext.setPlanningError(
-              "The second argument '%s' to function '%s' is not a number",
-              rexNodes.get(1),
+              "The third argument '%s' to function '%s' is not a boolean",
+              rexNodes.get(2),
               aggregateCall.getName()
           );
```

Comment:
this is a bit too late... a check like this should happen during validation and not during aggregator production; a straight `DruidException` might also be better.

Reply:
removed the code as early validation is added